From b29d9576aab682a227402057d98cf76b1eeae944 Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Tue, 17 Nov 2015 14:08:43 -0800
Subject: [PATCH 03/13] Don't attempt to convert degree to orientation enum
 twice

* Exif.getOrientation already calls getRotationForOrientationValues,
  which returns a degree value. Don't attempt to convert that degree
  value back into degrees, that makes no sense.

Change-Id: Ie35a0683e2d36801a9130ec19e1b3aa1f48b92c2
---
 .../processing/imagebackend/TaskCompressImageToJpeg.java       | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/com/android/camera/processing/imagebackend/TaskCompressImageToJpeg.java b/src/com/android/camera/processing/imagebackend/TaskCompressImageToJpeg.java
index 2e5976c46..e1d1689cc 100644
--- a/src/com/android/camera/processing/imagebackend/TaskCompressImageToJpeg.java
+++ b/src/com/android/camera/processing/imagebackend/TaskCompressImageToJpeg.java
@@ -95,8 +95,7 @@ public class TaskCompressImageToJpeg extends TaskJpegEncode {
      */
     public Map<Integer, Integer> exifGetMinimalTags(ExifInterface exif) {
         Map<Integer, Integer> map = new HashMap<>();
-        map.put(ExifInterface.TAG_ORIENTATION,
-                ExifInterface.getRotationForOrientationValue((short) Exif.getOrientation(exif)));
+        map.put(ExifInterface.TAG_ORIENTATION, Exif.getOrientation(exif));
         map.put(ExifInterface.TAG_PIXEL_X_DIMENSION, exif.getTagIntValue(
                 ExifInterface.TAG_PIXEL_X_DIMENSION));
         map.put(ExifInterface.TAG_PIXEL_Y_DIMENSION, exif.getTagIntValue(
-- 
2.17.1

